lib.rs

```diff
@@ -87,16 +140,24 @@ impl<D: Decorator> Format<D> {
         let r_decorator = self.decorator.decorate(record);
 
 
-        try!(self.print_msg_header(io, &r_decorator, record));
+        let mut skip_msg_comma = try!(self.print_msg_header(io, &r_decorator, record));
```

Comment:
I was making an example to show this, and realized that even if there wasn't a different between `Fn` and `FnOnce`, we still couldn't make it a member variable because `print_msg_header` takes `&self`, then `format_full` and `format_compact` take `&self`, and then you get to the impl of the `StreamFormat` trait, and that only gives `&self`.

So unless we use something like `Cell` to get interior mutability, we can't change `self` to set a member variable.